lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3679,6 +3619,49 @@ PotentialArchetype *GenericSignatureBuilder::realizePotentialArchetype(
   return pa;
 }
 
+static Type substituteConcreteType(GenericSignatureBuilder &builder,
+                                   PotentialArchetype *basePA,
+                                   TypeDecl *concreteDecl) {
+  if (!concreteDecl)
```

Comment:
Will we ever call it with a null `concreteDecl`? It seems like this should be an assert.

Reply:
Yeah, makes sense!